src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -134,6 +133,22 @@ export class FocusMonitor implements OnDestroy {
     this._windowFocusTimeoutId = setTimeout(() => this._windowFocused = false);
   }
 
+  /**
+   * Event listener for `focus` and 'blur' events on the document.
+   * Needs to be an arrow function in order to preserve the context when it gets bound.
+   */
+  private _documentFocusAndBlurListener = (event: FocusEvent) => {
+    const target = event.target as HTMLElement|null;
+    // We need to walk up the ancestor chain in order to support `checkChildren`.
+    for (let el = target; el; el = el.parentElement) {
```

Comment:
Idea to explore later: we would need to test whether this had a positive impact, but you could _potentially_ set an attribute to monitored elements to `break` out of this early (stopping on the first monitored element). Whether this improves anything depends on a lot of factors- depth of the DOM, depth of focus-monitored elements, number of focuses that _aren't_ monitored, and cost of reading the attribute.

Reply:
I'm not sure I follow...
You can monitor nested elements with `checkChildren` set to true, so it's not possible to break out early. Having an attribute on monitored elements is functionally equivalent to looking them up in a map.

Reply:
Yeah, it's equivalent to looking up in a map (which I forgot this class already has). My thinking was that we can stop this upward traversal once you hit any monitored element. That wouldn't trigger the behavior in the case of _nested_ monitored elements, but I'm not sure nested controls is a valid use case. Overall just idle musing on my part, no action necessary